VENDORING.md

```diff
@@ -0,0 +1,238 @@
+  name = "github.com/foo/bar"
+  branch = "fork-work-dev"
+  source = "github.com/myfork/bar"
+```
+
+__Note:__ Constraint a package is important because this is the only way to
+make sure about the version of an imported package. This prevents from any
+build failure that could result from a package change on the upstream.
+
+### Update a vendored package
+
+This case describes a situation where the developer adds some code relying on
+another version of a package already vendored. This is a very simple case since
+the package is already part of the vendoring. The developer will have to modify
+`Gopkg.toml` by updating the vendoring version:
+
+_Original file_
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  revision = "f5742cb6"
+```
+
+_Updated file_
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  revision = "a4be45c7"
+```
+
+And let `dep` update the vendoring by relying on this new revision:
+```bash
+dep ensure
+dep prune
+```
+
+### Remove a vendored package
+
+This case describes a situation where the developer modifies the existing
+code, no longer relying on a vendored package. Running the usual commands
+will remove the vendored package from `vendor` directory and from
+`Gopkg.lock`:
+```bash
+dep ensure
+dep prune
+```
+
+If a constraint has been set in `Gopkg.toml`, this is the responsibility of
+developer to remove it manually from this file.
+
+### Test a pending pull request
+
+This case describes a situation where the developer needs to perform some
+testing on his code, depending on a pending pull request from a vendored
+package. The easiest and safest way to do that is to rely on a feature proposed
+by `dep` called `override`. This allow the developer to leave the constraints
+unmodified but to rely on a different version of the vendored package for
+testing purpose.
+
+#### From the main repository
+
+The pending pull request can be submitted to the main repository, through a
+development branch `work-dev`, in which case the developer will have to modify
+`Gopkg.toml` as follow:
+_Original file_
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  revision = "f5742cb6"
+```
+
+_Updated file_
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  revision = "f5742cb6"
+
+[[override]]
+  name = "github.com/foo/bar"
+  branch = "work-dev"
+```
+
+#### From a forked repository
+
+The pending pull request can be submitted to a forked repository, through
+a development branch called `work-dev`, in which case the developer will have
+to modify `Gopkg.toml` as follow:
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  revision = "f5742cb6"
+```
+
+_Updated file_
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  revision = "f5742cb6"
+
+[[override]]
+  name = "github.com/foo/bar"
+  branch = "work-dev"
+  source = "github.com/myfork/bar"
+```
+
+It is more common to see this case happening. Usually, developers don't have
+permissions to push directly to the main repository.
```

Comment:
Last point: please could you add a reference to this document to https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#re-vendor-prs.

Reply:
done